Prefer xref over code-style

mikekistler opened the pull request:
This PR converts code-style references to .NET classes to xref references.

Prior to this there was a mix of code-style and xref references in this file.

<!-- PREVIEW-TABLE-START -->

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/fundamentals/openapi/aspnetcore-openapi.md](https://github.com/dotnet/AspNetCore.Docs/blob/69ad691c81a1c7add04b682218d5c761fe426488/aspnetcore/fundamentals/openapi/aspnetcore-openapi.md) | [aspnetcore/fundamentals/openapi/aspnetcore-openapi](https://review.learn.microsoft.com/en-us/aspnet/core/fundamentals/openapi/aspnetcore-openapi?branch=pr-en-us-33853) |


<!-- PREVIEW-TABLE-END -->

tdykstra commented on aspnetcore/fundamentals/openapi/aspnetcore-openapi.md:
You can use xref links and still get the rendering with brackets and without "Attribute". For example, instead of
`<xref:Microsoft.AspNetCore.Mvc.ProducesAttribute>`
which renders as ProducesAttribute
you can have
`[[Produces]](xref:Microsoft.AspNetCore.Mvc.ProducesAttribute)`
which renders as [Produces]

It's not necessary, I'm just mentioning it as an option.

mikekistler replied:
Discussed with @captainsafia and we decided that the style you describe is better / more consistent with the rest of the docs, so I'll convert all the links to that.

tdykstra reviewed (approved):
Looks good, I mention one optional format.

mikekistler commented:
@tdykstra @captainsafia I've switched the links to use a style like Tom suggested, but I had added one more stylistic twist. In cases where the link was to an attribute and was followed by "attribute", I moved "attribute" inside the link text. Likewise for "extension method". Here's a snip that shows how this looks now:

![image](https://github.com/user-attachments/assets/e7ef3280-a4d8-42fc-a6a9-600a3bee36ea)

I don't have strong feelings about this but it seems worth trying. Let me know if you like it better the other way and I can easily change it.

tdykstra commented:
> I don't have strong feelings about this but it seems worth trying.

I don't either, so I'd be interested in getting input from @captainsafia and @Rick-Anderson

mikekistler commented:
@captainsafia convinced me that we should not put "attribute" or "extension method" inside the link text, so I just pushed another commit that reverts that stylistic twist.

I think this is good to go now. @captainsafia can you please give it a quick review & approval and then I will merge.

guardrex commented:
@mikekistler ... Please squash your PRs when merging.

mikekistler commented:
@guardrex Apologies -- I will try to remember to do this next time.

guardrex commented:
No worries ... just keeping the history tidy 😄 .
